addon/components/pie-chart.js

```diff
@@ -227,12 +238,40 @@ const PieChartComponent = ChartComponent.extend(FloatingTooltipMixin,
   numSlices: Ember.computed.alias('finishedData.length'),
 
   // Offset slices so that the largest slice finishes at 12 o'clock
-  startOffset: Ember.computed('finishedData', function() {
-    var data = this.get('finishedData');
-    var sum = data.reduce(function(p, d) {
+  //
+  // Rotate the pie and concentrate all small slices at 8 to 10 o'clock if
+  // there is a high density of small slices inside the pie.
+  startOffset: Ember.computed('finishedData', 'sortKey', function() {
+    var finishedData = this.get('finishedData');
+    var sum = finishedData.reduce(function(p, d) {
       return d.percent + p;
     }, 0);
-    return _.last(data).percent / sum * 2 * Math.PI;
+
+    // If there is a high concentration of small slices inside the pie,
+    // we want start the pie at a "safe" offset
+    var detectDenseSmallSlices = function (finishedData) {
+      var sortedData = _.sortBy(finishedData, "percent");
+      var numSmallSlices = 2;
+      var startInd = 0;
+      var endInd = Math.min(numSmallSlices, sortedData.length);
+
+      var averageSmallSlicesPercent = sortedData.slice(startInd, endInd).reduce(function(p,d) {
+        return d.percent / Math.max(1, endInd - startInd) + p;
+      }, 0);
+
+      if (averageSmallSlicesPercent <= 2.75) {
+        return true;
+      } else if ((averageSmallSlicesPercent <= 5) && (45 <=_.last(sortedData).percent) && (_.last(sortedData).percent <= 55)) {
```

Comment:
all of these complicated condition deserve a better name/variable for each part itself and/or a comment explaining.